commands/create/kubeconfig/command.go

```diff
@@ -367,28 +371,48 @@ func createKubeconfig(ctx context.Context, args Arguments) (createKubeconfigResu
 	auxParams := clientWrapper.DefaultAuxiliaryParams()
 	auxParams.ActivityName = createKubeconfigActivityName
 
-	// get cluster details
-	clusterDetailsResponse, err := clientWrapper.GetClusterV4(args.clusterID, auxParams)
+	// Try v5 first, then fall back to v4.
+	if args.verbose {
+		fmt.Println(color.WhiteString("Fetching cluster details using the v5 API endpoint"))
+	}
+	clusterDetailsResponseV5, err := clientWrapper.GetClusterV5(args.clusterID, auxParams)
 	if err != nil {
-		// TODO: return properly typed errors
-		if clientErr, ok := err.(*clienterror.APIError); ok {
-			return result, microerror.Maskf(clientErr,
-				fmt.Sprintf("HTTP Status: %d, %s", clientErr.HTTPStatusCode, clientErr.ErrorMessage))
+		if clienterror.IsNotFoundError(err) {
+			// We ignore a 404 here and try v4 next.
+			if args.verbose {
+				fmt.Println(color.WhiteString("Cluster not found via the v5 endpoint"))
+			}
+		} else {
+			// For all other errors than 404, we fail.
+			return result, microerror.Mask(err)
 		}
-
-		return result, microerror.Mask(err)
+	} else {
+		result.apiEndpoint = clusterDetailsResponseV5.Payload.APIEndpoint
 	}
 
-	var apiEndpoint string
-	{
-		if args.tenantInternal {
-			baseEndpoint := strings.Split(clusterDetailsResponse.Payload.APIEndpoint, urlDelimiter)[1:]
-			apiEndpoint = fmt.Sprintf("https://%s.%s", tenantInternalAPIPrefix, strings.Join(baseEndpoint, urlDelimiter))
-		} else {
-			apiEndpoint = clusterDetailsResponse.Payload.APIEndpoint
+	if result.apiEndpoint == "" {
```

Comment:
Would it make sense to extract this to separate function and inline the call in if-case above? I find this logic a little difficult to follow, because first it works with V5, then if that fails it potentially just logs and next step is suddenly v4 call if some variable is unset so far.

I feel like these generated types require some wrapping so that the higher level business logic would be clean, but is a bit overkill at this point maybe.

Reply:
Changed the way cluster details are pulled.